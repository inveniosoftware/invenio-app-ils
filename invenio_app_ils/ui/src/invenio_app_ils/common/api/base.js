import axios from 'axios';
import { sessionManager } from '../../authentication/services';

const token = sessionManager.token ? sessionManager.token.value : '';

const apiConfig = {
  baseURL:
    process.env.NODE_ENV === 'production'
      ? '/api'
      : 'https://127.0.0.1:5000/api',
  headers: {
    Authorization: `bearer ${token}`,
  },
  withCredentials: true,
};

const http = axios.create(apiConfig);

export { http, apiConfig };
