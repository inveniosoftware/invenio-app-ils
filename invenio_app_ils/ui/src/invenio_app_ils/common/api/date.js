import { DateTime } from 'luxon';
import _get from 'lodash/get';

export const fromISO = stringDate => {
  return DateTime.fromISO(stringDate);
};

/**
 * Converts datetime JSON string to luxon Datetime object
 * @param date
 * @returns {DateTime}
 */
export const toISO = date => {
  return date.toISO();
};

/**
 *  Serializes date for display in short format
 *  @param {DateTime} date luxon DateTime
 *  @re
 */
export const toShortDateTime = date => {
  return date.toFormat('yyyy-MM-dd HH:mm');
};

/**
 *  Serializes date for display in short format
 *  @param {DateTime} date luxon DateTime
 *  @re
 */
export const toShortDate = date => {
  return date.toFormat('yyyy-MM-dd');
};

/**
 *  Serializes date for UTC date in short format used to
 *  normalise dates for REST API
 *  @param {DateTime} date luxon DateTime
 *  @re
 */
export const toUTCShortDate = date => {
  if (!(typeof date === 'string' || date instanceof DateTime)) {
    throw Error('Wrong date type provided');
  }
  if (typeof date === 'string') {
    date = fromISO(date);
  }
  return date.toUTC().toFormat('yyyy-MM-dd');
};

/**
 *  Serializes date for display in tables
 */
export function dateFormatter({ col, row }) {
  const dateField = _get(row, col.field);
  return dateField ? toShortDate(fromISO(dateField)) : null;
}
