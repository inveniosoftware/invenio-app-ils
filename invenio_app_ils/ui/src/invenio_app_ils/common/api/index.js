export { document } from './documents/document';
export { http } from './base';
export { item } from './items/item';
export { eitem } from './eitems/eitem';
export { loan } from './loans/loan';
export { patron } from './patrons/patron';
export { location } from './locations/location';
export { internalLocation } from './locations/internalLocation';
export { keyword } from './keywords/keyword';
export { series } from './series/series';
export { related } from './related/related';
