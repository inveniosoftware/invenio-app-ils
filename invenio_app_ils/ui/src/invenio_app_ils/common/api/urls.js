export const ApiURLS = {
  loans: {
    loan: '/circulation/loans/:loanPid',
    list: '/circulation/loans/',
    checkout: '/circulation/loans/checkout',
    request: '/circulation/loans/request',
  },
  items: {},
  documents: {},
  locations: {},
  users: {},
};
