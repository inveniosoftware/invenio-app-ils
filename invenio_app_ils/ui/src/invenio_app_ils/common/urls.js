import { generatePath } from 'react-router-dom';
import { loan as loanApi } from './api/loan';
import { item as itemApi } from './api/item';

export const FrontSiteURLS = {};

const BackOfficeBase = '/backoffice';

export const BackOfficeURLS = {
  home: BackOfficeBase,
  documentDetails: `${BackOfficeBase}/documents/:documentPid`,
  itemsSearch: `${BackOfficeBase}/items`,
  itemDetails: `${BackOfficeBase}/items/:itemPid`,
  loansSearch: `${BackOfficeBase}/loans`,
  loanDetails: `${BackOfficeBase}/loans/:loanPid`,
  patronDetails: `${BackOfficeBase}/users/:userPid`,
};

export const viewLoanDetailsUrl = loanPid => {
  return generatePath(BackOfficeURLS.loanDetails, {
    loanPid: loanPid,
  });
};

export const showAllLoansUrl = (
  documentPid,
  itemPid,
  state,
  patronPid,
  extraQuery
) => {
  const qs = loanApi.buildLoansQuery(
    documentPid,
    itemPid,
    state,
    patronPid,
    extraQuery
  );
  return `${BackOfficeURLS.loansSearch}?q=${qs}`;
};

export const viewItemDetailsUrl = itemPid => {
  return generatePath(BackOfficeURLS.itemDetails, {
    itemPid: itemPid,
  });
};

export const showAllItemsUrl = (documentPid, state, extraQuery) => {
  const qs = itemApi.buildItemsQuery(documentPid, state, extraQuery);
  return `${BackOfficeURLS.itemsSearch}?q=${qs}`;
};
