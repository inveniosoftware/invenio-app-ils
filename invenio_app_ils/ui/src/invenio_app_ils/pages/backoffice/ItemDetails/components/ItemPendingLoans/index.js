import { compose } from 'redux';
import { connect } from 'react-redux';
import { withRouter } from 'react-router-dom';
import {
  fetchPendingLoans,
  pendingLoansChangeSortBy,
  pendingLoansChangeSortOrder,
} from './state/actions';
import ItemPendingLoansComponent from './ItemPendingLoans';

const mapStateToProps = state => ({
  data: state.itemPendingLoans.data,
  isLoading: state.itemPendingLoans.isLoading,
  hasError: state.itemPendingLoans.hasError,
  currentSortBy: state.itemPendingLoans.sortBy,
  currentSortOrder: state.itemPendingLoans.sortOrder,
});

const mapDispatchToProps = dispatch => ({
  fetchPendingLoans: (documentPid, itemPid) =>
    dispatch(fetchPendingLoans(documentPid, itemPid)),
  pendingLoansChangeSortBy: (documentPid, itemPid) =>
    dispatch(pendingLoansChangeSortBy(documentPid, itemPid)),
  pendingLoansChangeSortOrder: (documentPid, itemPid) =>
    dispatch(pendingLoansChangeSortOrder(documentPid, itemPid)),
});

export const ItemPendingLoans = compose(
  withRouter,
  connect(
    mapStateToProps,
    mapDispatchToProps
  )
)(ItemPendingLoansComponent);
