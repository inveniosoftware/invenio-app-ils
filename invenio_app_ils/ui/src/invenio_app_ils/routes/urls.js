import { generatePath } from 'react-router-dom';
import { invenioConfig } from '../common/config';

const FrontSiteBase = '/';

const FrontSiteRoutesList = {
  home: FrontSiteBase,
  patronProfile: `${FrontSiteBase}profile`,
  documentsList: `${FrontSiteBase}search`,
  documentDetails: `${FrontSiteBase}records/:documentPid`,
};

const FrontSiteRoutesGenerators = {
  documentsListWithQuery: qs => `${FrontSiteRoutesList.documentsList}?q=${qs}`,
  documentDetailsFor: documentPid =>
    generatePath(FrontSiteRoutesList.documentDetails, {
      documentPid: documentPid,
    }),
};

export const FrontSiteRoutes = {
  ...FrontSiteRoutesGenerators,
  ...FrontSiteRoutesList,
};

const BackOfficeBase = '/backoffice';

const BackOfficeRoutesList = {
  home: BackOfficeBase,
  documentsList: `${BackOfficeBase}/documents`,
  documentsCreate: `${BackOfficeBase}/documents/create`,
  documentDetails: `${BackOfficeBase}/documents/:documentPid`,
  eitemsList: `${BackOfficeBase}/eitems`,
  eitemDetails: `${BackOfficeBase}/eitems/:eitemPid`,
  itemsList: `${BackOfficeBase}/items`,
  itemDetails: `${BackOfficeBase}/items/:itemPid`,
  loansList: `${BackOfficeBase}/loans`,
  loanDetails: `${BackOfficeBase}/loans/:loanPid`,
  patronsList: `${BackOfficeBase}/patrons`,
  patronDetails: `${BackOfficeBase}/patrons/:patronPid`,
  locationsList: `${BackOfficeBase}/locations`,
  seriesList: `${BackOfficeBase}/series`,
  seriesDetails: `${BackOfficeBase}/series/:seriesPid`,
};

const BackOfficeRouteGenerators = {
  documentsListWithQuery: qs => `${BackOfficeRoutesList.documentsList}?q=${qs}`,
  documentDetailsFor: documentPid =>
    generatePath(BackOfficeRoutesList.documentDetails, {
      documentPid: documentPid,
    }),
  eitemDetailsFor: eitemPid =>
    generatePath(BackOfficeRoutesList.eitemDetails, { eitemPid: eitemPid }),
  itemsListWithQuery: qs => `${BackOfficeRoutesList.itemsList}?q=${qs}`,
  itemDetailsFor: itemPid =>
    generatePath(BackOfficeRoutesList.itemDetails, { itemPid: itemPid }),
  loansListWithQuery: qs => `${BackOfficeRoutesList.loansList}?q=${qs}`,
  loanDetailsFor: loanPid =>
    generatePath(BackOfficeRoutesList.loanDetails, { loanPid: loanPid }),
  patronDetailsFor: patronPid =>
    generatePath(BackOfficeRoutesList.patronDetails, { patronPid: patronPid }),
  seriesListWithQuery: qs => `${BackOfficeRoutesList.seriesList}?q=${qs}`,
  seriesDetailsFor: seriesPid =>
    generatePath(BackOfficeRoutesList.seriesDetails, { seriesPid: seriesPid }),
};

export const BackOfficeRoutes = {
  ...BackOfficeRoutesList,
  ...BackOfficeRouteGenerators,
};

export const openRecordEditor = (path, recid = '') => {
  window.open(`${invenioConfig.editor.url}${path}${recid}`, '_blank');
};
