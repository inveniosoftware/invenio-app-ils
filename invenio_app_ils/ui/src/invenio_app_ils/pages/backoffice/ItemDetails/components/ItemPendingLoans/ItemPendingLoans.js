import React, { Component } from 'react';
import PropTypes from 'prop-types';
import { Loader, Error } from '../../../../../common/components';
import { toString } from '../../../../../common/api/date';
import { ResultsTable } from '../../../../../common/components';

import './ItemPendingLoans.scss';
import {
  showAllLoansUrl,
  viewLoanDetailsUrl,
} from '../../../../../common/urls';

export default class ItemPendingLoans extends Component {
  constructor(props) {
    super(props);
    this.fetchPendingLoans = props.fetchPendingLoans;
    this.showDetailsUrl = viewLoanDetailsUrl;
    this.showAllUrl = showAllLoansUrl;
  }

  componentDidMount() {
    const { document_pid, item_pid } = this.props.item.metadata;
    this.fetchPendingLoans(document_pid, item_pid);
  }

  _getFormattedDate = d => (d ? toString(d) : '');

  _showDetailsHandler = loan_pid =>
    this.props.history.push(this.showDetailsUrl(loan_pid));

  _showAllHandler = () => {
    const { document_pid, item_pid } = this.props.item.metadata;
    this.props.history.push(
      this.showAllUrl(document_pid, item_pid, 'PENDING', null)
    );
  };

  prepareData() {
    return this.props.data.map(row => ({
      ID: row.loan_pid,
      'Patron PID': row.patron_pid,
      'Request created': this._getFormattedDate(row.updated),
    }));
  }

  render() {
    const rows = this.prepareData();
    const { data, isLoading, hasError } = this.props;
    const errorData = hasError ? data : null;
    return (
      <Loader isLoading={isLoading}>
        <Error error={errorData}>
          <ResultsTable
            rows={rows}
            name={'Pending loans requests'}
            detailsClickHandler={this._showDetailsHandler}
            showAllClickHandler={{
              handler: this._showAllHandler,
              params: null,
            }}
            showMaxRows={this.props.showMaxPendingLoans}
          />
        </Error>
      </Loader>
    );
  }
}

ItemPendingLoans.propTypes = {
  item: PropTypes.object.isRequired,
  fetchPendingLoans: PropTypes.func.isRequired,
  data: PropTypes.array.isRequired,
  showMaxPendingLoans: PropTypes.number,
};

ItemPendingLoans.defaultProps = {
  showMaxPendingLoans: 5,
};
