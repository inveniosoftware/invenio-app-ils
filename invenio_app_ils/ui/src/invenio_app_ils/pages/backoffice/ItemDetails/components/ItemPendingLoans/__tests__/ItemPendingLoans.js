import React from 'react';
import { shallow, mount } from 'enzyme';
import { generatePath } from 'react-router';
import { Settings } from 'luxon';
import { fromISO } from '../../../../../../common/api/date';
import { BackOfficeURLS } from '../../../../../../common/urls';
import ItemPendingLoans from '../ItemPendingLoans';

Settings.defaultZoneName = 'utc';
const d = fromISO('2018-01-01T11:05:00+01:00');

describe('ItemPendingLoans tests', () => {
  let component;
  afterEach(() => {
    if (component) {
      component.unmount();
    }
  });

  const item = {
    metadata: {
      document_pid: 111,
      item_pid: 222,
    },
  };

  it('should load the details component', () => {
    const component = shallow(
      <ItemPendingLoans
        item={item}
        history={() => {}}
        data={[]}
        fetchPendingLoans={() => {}}
      />
    );
    expect(component).toMatchSnapshot();
  });

  it('should fetch pending loans on mount', () => {
    const mockedFetchPendingLoans = jest.fn();
    component = mount(
      <ItemPendingLoans
        item={item}
        history={() => {}}
        data={[]}
        fetchPendingLoans={mockedFetchPendingLoans}
      />
    );
    expect(mockedFetchPendingLoans).toHaveBeenCalledWith(item.item_pid);
  });

  it('should render show a message with no pending loans', () => {
    component = mount(
      <ItemPendingLoans
        item={item}
        history={() => {}}
        data={[]}
        fetchPendingLoans={() => {}}
      />
    );

    expect(component).toMatchSnapshot();
    const message = component
      .find('Message')
      .filterWhere(element => element.prop('data-test') === 'no-results');
    expect(message).toHaveLength(1);
  });

  it('should render pending loans', () => {
    const data = [
      {
        loan_pid: 'loan1',
        patron_pid: 'patron_1',
        updated: d,
        created: d,
        start_date: d,
        end_date: d,
      },
      {
        loan_pid: 'loan2',
        patron_pid: 'patron_2',
        updated: d,
        created: d,
        start_date: d,
        end_date: d,
      },
    ];

    component = mount(
      <ItemPendingLoans
        item={item}
        history={() => {}}
        data={data}
        fetchPendingLoans={() => {}}
      />
    );

    expect(component).toMatchSnapshot();
    const rows = component
      .find('TableRow')
      .filterWhere(
        element =>
          element.prop('data-test') === 'loan1' ||
          element.prop('data-test') === 'loan2'
      );
    expect(rows).toHaveLength(2);

    const footer = component
      .find('TableRow')
      .filterWhere(element => element.prop('data-test') === 'footer');
    expect(footer).toHaveLength(0);
  });

  it('should render the show all button when showing only a few pending loans', () => {
    const data = [
      {
        loan_pid: 'loan1',
        patron_pid: 'patron_1',
        updated: d,
        created: d,
        start_date: d,
        end_date: d,
      },
      {
        loan_pid: 'loan2',
        patron_pid: 'patron_2',
        updated: d,
        created: d,
        start_date: d,
        end_date: d,
      },
    ];

    component = mount(
      <ItemPendingLoans
        item={item}
        history={() => {}}
        data={data}
        fetchPendingLoans={() => {}}
        showMaxPendingLoans={1}
      />
    );

    expect(component).toMatchSnapshot();
    const footer = component
      .find('TableFooter')
      .filterWhere(element => element.prop('data-test') === 'footer');
    expect(footer).toHaveLength(1);
  });

  it('should go to loan details when clicking on a pending loan', () => {
    const mockedHistoryPush = jest.fn();
    const historyFn = {
      push: mockedHistoryPush,
    };

    const data = [
      {
        loan_pid: 'loan1',
        patron_pid: 'patron_1',
        updated: d,
        created: d,
        start_date: d,
        end_date: d,
      },
    ];

    component = mount(
      <ItemPendingLoans
        item={item}
        history={historyFn}
        data={data}
        fetchPendingLoans={() => {}}
        showMaxPendingLoans={1}
      />
    );

    const firstId = data[0].loan_pid;
    const button = component
      .find('TableRow')
      .filterWhere(element => element.prop('data-test') === firstId)
      .find('i');
    button.simulate('click');

    const expectedParam = generatePath(BackOfficeURLS.loanDetails, {
      loanPid: firstId,
    });
    expect(mockedHistoryPush).toHaveBeenCalledWith(expectedParam);
  });
});
