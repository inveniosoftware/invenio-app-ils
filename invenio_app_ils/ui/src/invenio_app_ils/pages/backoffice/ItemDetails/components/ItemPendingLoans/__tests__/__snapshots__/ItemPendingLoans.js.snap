// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ItemPendingLoans tests should load the details component 1`] = `
<Loader>
  <Error
    error={null}
  >
    <ResultsTable
      actionClickHandler={[Function]}
      actionComponent={null}
      name="Pending loans requests"
      rows={Array []}
      showAllClickHandler={
        Object {
          "handler": [Function],
          "params": null,
        }
      }
      showMaxRows={5}
    />
  </Error>
</Loader>
`;

exports[`ItemPendingLoans tests should render pending loans 1`] = `
<ItemPendingLoans
  data={
    Array [
      Object {
        "created": "2018-01-01T10:05:00.000Z",
        "end_date": "2018-01-01T10:05:00.000Z",
        "loan_pid": "loan1",
        "patron_pid": "patron_1",
        "start_date": "2018-01-01T10:05:00.000Z",
        "updated": "2018-01-01T10:05:00.000Z",
      },
      Object {
        "created": "2018-01-01T10:05:00.000Z",
        "end_date": "2018-01-01T10:05:00.000Z",
        "loan_pid": "loan2",
        "patron_pid": "patron_2",
        "start_date": "2018-01-01T10:05:00.000Z",
        "updated": "2018-01-01T10:05:00.000Z",
      },
    ]
  }
  fetchPendingLoans={[Function]}
  history={[Function]}
  item={
    Object {
      "metadata": Object {
        "document_pid": 111,
        "item_pid": 222,
      },
    }
  }
  showMaxPendingLoans={5}
>
  <Loader>
    <Error
      error={null}
    >
      <ResultsTable
        actionClickHandler={[Function]}
        actionComponent={null}
        name="Pending loans requests"
        rows={
          Array [
            Object {
              "ID": "loan1",
              "Patron ID": "patron_1",
              "Request created": "2018-01-01 10:05",
              "State": undefined,
            },
            Object {
              "ID": "loan2",
              "Patron ID": "patron_2",
              "Request created": "2018-01-01 10:05",
              "State": undefined,
            },
          ]
        }
        showAllClickHandler={
          Object {
            "handler": [Function],
            "params": null,
          }
        }
        showMaxRows={5}
      >
        <Segment>
          <div
            className="ui segment"
          >
            <Header
              as="h3"
            >
              <h3
                className="ui header"
              >
                Pending loans requests
              </h3>
            </Header>
            <Table
              as="table"
              selectable={true}
              striped={true}
            >
              <table
                className="ui selectable striped table"
              >
                <ResultsTableHeader
                  columns={
                    Array [
                      "ID",
                      "Request created",
                      "Patron ID",
                      "State",
                    ]
                  }
                >
                  <TableHeader
                    as="thead"
                  >
                    <thead
                      className=""
                    >
                      <TableRow
                        as="tr"
                        cellAs="td"
                        data-test="header"
                      >
                        <tr
                          className=""
                          data-test="header"
                        >
                          <TableHeaderCell
                            as="th"
                            collapsing={true}
                            width={2}
                          >
                            <TableCell
                              as="th"
                              className=""
                              collapsing={true}
                              width={2}
                            >
                              <th
                                className="collapsing two wide"
                              />
                            </TableCell>
                          </TableHeaderCell>
                          <TableHeaderCell
                            as="th"
                            key="ID"
                            width={2}
                          >
                            <TableCell
                              as="th"
                              className=""
                              width={2}
                            >
                              <th
                                className="two wide"
                              >
                                ID
                              </th>
                            </TableCell>
                          </TableHeaderCell>
                          <TableHeaderCell
                            as="th"
                            key="Request created"
                          >
                            <TableCell
                              as="th"
                              className=""
                            >
                              <th
                                className=""
                              >
                                Request created
                              </th>
                            </TableCell>
                          </TableHeaderCell>
                          <TableHeaderCell
                            as="th"
                            key="Patron ID"
                          >
                            <TableCell
                              as="th"
                              className=""
                            >
                              <th
                                className=""
                              >
                                Patron ID
                              </th>
                            </TableCell>
                          </TableHeaderCell>
                          <TableHeaderCell
                            as="th"
                            key="State"
                          >
                            <TableCell
                              as="th"
                              className=""
                            >
                              <th
                                className=""
                              >
                                State
                              </th>
                            </TableCell>
                          </TableHeaderCell>
                        </tr>
                      </TableRow>
                    </thead>
                  </TableHeader>
                </ResultsTableHeader>
                <ResultsTableBody
                  actionClickHandler={[Function]}
                  actionComponent={null}
                  columns={
                    Array [
                      "ID",
                      "Request created",
                      "Patron ID",
                      "State",
                    ]
                  }
                  rows={
                    Array [
                      Object {
                        "ID": "loan1",
                        "Patron ID": "patron_1",
                        "Request created": "2018-01-01 10:05",
                        "State": undefined,
                      },
                      Object {
                        "ID": "loan2",
                        "Patron ID": "patron_2",
                        "Request created": "2018-01-01 10:05",
                        "State": undefined,
                      },
                    ]
                  }
                >
                  <TableBody
                    as="tbody"
                  >
                    <tbody
                      className=""
                    >
                      <TableRow
                        as="tr"
                        cellAs="td"
                        data-test="loan1"
                        key="loan1"
                      >
                        <tr
                          className=""
                          data-test="loan1"
                        >
                          <TableCell
                            as="td"
                            key="details"
                            textAlign="center"
                          >
                            <td
                              className="center aligned"
                            >
                              <Button
                                as="button"
                                circular={true}
                                compact={true}
                                icon="eye"
                                onClick={[Function]}
                              >
                                <button
                                  className="ui circular compact icon button"
                                  onClick={[Function]}
                                >
                                  <Icon
                                    as="i"
                                    name="eye"
                                  >
                                    <i
                                      aria-hidden="true"
                                      className="eye icon"
                                    />
                                  </Icon>
                                </button>
                              </Button>
                            </td>
                          </TableCell>
                          <TableCell
                            as="td"
                            data-test="ID-loan1"
                            key="0-loan1"
                          >
                            <td
                              className=""
                              data-test="ID-loan1"
                            >
                              loan1
                            </td>
                          </TableCell>
                          <TableCell
                            as="td"
                            data-test="Request created-loan1"
                            key="1-loan1"
                          >
                            <td
                              className=""
                              data-test="Request created-loan1"
                            >
                              2018-01-01 10:05
                            </td>
                          </TableCell>
                          <TableCell
                            as="td"
                            data-test="Patron ID-loan1"
                            key="2-loan1"
                          >
                            <td
                              className=""
                              data-test="Patron ID-loan1"
                            >
                              patron_1
                            </td>
                          </TableCell>
                          <TableCell
                            as="td"
                            data-test="State-loan1"
                            key="3-loan1"
                          >
                            <td
                              className=""
                              data-test="State-loan1"
                            >
                              -
                            </td>
                          </TableCell>
                        </tr>
                      </TableRow>
                      <TableRow
                        as="tr"
                        cellAs="td"
                        data-test="loan2"
                        key="loan2"
                      >
                        <tr
                          className=""
                          data-test="loan2"
                        >
                          <TableCell
                            as="td"
                            key="details"
                            textAlign="center"
                          >
                            <td
                              className="center aligned"
                            >
                              <Button
                                as="button"
                                circular={true}
                                compact={true}
                                icon="eye"
                                onClick={[Function]}
                              >
                                <button
                                  className="ui circular compact icon button"
                                  onClick={[Function]}
                                >
                                  <Icon
                                    as="i"
                                    name="eye"
                                  >
                                    <i
                                      aria-hidden="true"
                                      className="eye icon"
                                    />
                                  </Icon>
                                </button>
                              </Button>
                            </td>
                          </TableCell>
                          <TableCell
                            as="td"
                            data-test="ID-loan2"
                            key="0-loan2"
                          >
                            <td
                              className=""
                              data-test="ID-loan2"
                            >
                              loan2
                            </td>
                          </TableCell>
                          <TableCell
                            as="td"
                            data-test="Request created-loan2"
                            key="1-loan2"
                          >
                            <td
                              className=""
                              data-test="Request created-loan2"
                            >
                              2018-01-01 10:05
                            </td>
                          </TableCell>
                          <TableCell
                            as="td"
                            data-test="Patron ID-loan2"
                            key="2-loan2"
                          >
                            <td
                              className=""
                              data-test="Patron ID-loan2"
                            >
                              patron_2
                            </td>
                          </TableCell>
                          <TableCell
                            as="td"
                            data-test="State-loan2"
                            key="3-loan2"
                          >
                            <td
                              className=""
                              data-test="State-loan2"
                            >
                              -
                            </td>
                          </TableCell>
                        </tr>
                      </TableRow>
                    </tbody>
                  </TableBody>
                </ResultsTableBody>
                <ResultsTableFooter
                  allRowsNumber={2}
                  columnsNumber={4}
                  showAllClickHandler={
                    Object {
                      "handler": [Function],
                      "params": null,
                    }
                  }
                  showMaxRows={5}
                />
              </table>
            </Table>
          </div>
        </Segment>
      </ResultsTable>
    </Error>
  </Loader>
</ItemPendingLoans>
`;

exports[`ItemPendingLoans tests should render show a message with no pending loans 1`] = `
<ItemPendingLoans
  data={Array []}
  fetchPendingLoans={[Function]}
  history={[Function]}
  item={
    Object {
      "metadata": Object {
        "document_pid": 111,
        "item_pid": 222,
      },
    }
  }
  showMaxPendingLoans={5}
>
  <Loader>
    <Error
      error={null}
    >
      <ResultsTable
        actionClickHandler={[Function]}
        actionComponent={null}
        name="Pending loans requests"
        rows={Array []}
        showAllClickHandler={
          Object {
            "handler": [Function],
            "params": null,
          }
        }
        showMaxRows={5}
      >
        <Segment>
          <div
            className="ui segment"
          >
            <Header
              as="h3"
            >
              <h3
                className="ui header"
              >
                Pending loans requests
              </h3>
            </Header>
            <Message
              data-test="no-results"
            >
              <div
                className="ui message"
                data-test="no-results"
              >
                There are no 
                Pending loans requests
              </div>
            </Message>
          </div>
        </Segment>
      </ResultsTable>
    </Error>
  </Loader>
</ItemPendingLoans>
`;

exports[`ItemPendingLoans tests should render the show all button when showing only a few pending loans 1`] = `
<ItemPendingLoans
  data={
    Array [
      Object {
        "created": "2018-01-01T10:05:00.000Z",
        "end_date": "2018-01-01T10:05:00.000Z",
        "loan_pid": "loan1",
        "patron_pid": "patron_1",
        "start_date": "2018-01-01T10:05:00.000Z",
        "updated": "2018-01-01T10:05:00.000Z",
      },
      Object {
        "created": "2018-01-01T10:05:00.000Z",
        "end_date": "2018-01-01T10:05:00.000Z",
        "loan_pid": "loan2",
        "patron_pid": "patron_2",
        "start_date": "2018-01-01T10:05:00.000Z",
        "updated": "2018-01-01T10:05:00.000Z",
      },
    ]
  }
  fetchPendingLoans={[Function]}
  history={[Function]}
  item={
    Object {
      "metadata": Object {
        "document_pid": 111,
        "item_pid": 222,
      },
    }
  }
  showMaxPendingLoans={1}
>
  <Loader>
    <Error
      error={null}
    >
      <ResultsTable
        actionClickHandler={[Function]}
        actionComponent={null}
        name="Pending loans requests"
        rows={
          Array [
            Object {
              "ID": "loan1",
              "Patron ID": "patron_1",
              "Request created": "2018-01-01 10:05",
              "State": undefined,
            },
            Object {
              "ID": "loan2",
              "Patron ID": "patron_2",
              "Request created": "2018-01-01 10:05",
              "State": undefined,
            },
          ]
        }
        showAllClickHandler={
          Object {
            "handler": [Function],
            "params": null,
          }
        }
        showMaxRows={1}
      >
        <Segment>
          <div
            className="ui segment"
          >
            <Header
              as="h3"
            >
              <h3
                className="ui header"
              >
                Pending loans requests
              </h3>
            </Header>
            <Table
              as="table"
              selectable={true}
              striped={true}
            >
              <table
                className="ui selectable striped table"
              >
                <ResultsTableHeader
                  columns={
                    Array [
                      "ID",
                      "Request created",
                      "Patron ID",
                      "State",
                    ]
                  }
                >
                  <TableHeader
                    as="thead"
                  >
                    <thead
                      className=""
                    >
                      <TableRow
                        as="tr"
                        cellAs="td"
                        data-test="header"
                      >
                        <tr
                          className=""
                          data-test="header"
                        >
                          <TableHeaderCell
                            as="th"
                            collapsing={true}
                            width={2}
                          >
                            <TableCell
                              as="th"
                              className=""
                              collapsing={true}
                              width={2}
                            >
                              <th
                                className="collapsing two wide"
                              />
                            </TableCell>
                          </TableHeaderCell>
                          <TableHeaderCell
                            as="th"
                            key="ID"
                            width={2}
                          >
                            <TableCell
                              as="th"
                              className=""
                              width={2}
                            >
                              <th
                                className="two wide"
                              >
                                ID
                              </th>
                            </TableCell>
                          </TableHeaderCell>
                          <TableHeaderCell
                            as="th"
                            key="Request created"
                          >
                            <TableCell
                              as="th"
                              className=""
                            >
                              <th
                                className=""
                              >
                                Request created
                              </th>
                            </TableCell>
                          </TableHeaderCell>
                          <TableHeaderCell
                            as="th"
                            key="Patron ID"
                          >
                            <TableCell
                              as="th"
                              className=""
                            >
                              <th
                                className=""
                              >
                                Patron ID
                              </th>
                            </TableCell>
                          </TableHeaderCell>
                          <TableHeaderCell
                            as="th"
                            key="State"
                          >
                            <TableCell
                              as="th"
                              className=""
                            >
                              <th
                                className=""
                              >
                                State
                              </th>
                            </TableCell>
                          </TableHeaderCell>
                        </tr>
                      </TableRow>
                    </thead>
                  </TableHeader>
                </ResultsTableHeader>
                <ResultsTableBody
                  actionClickHandler={[Function]}
                  actionComponent={null}
                  columns={
                    Array [
                      "ID",
                      "Request created",
                      "Patron ID",
                      "State",
                    ]
                  }
                  rows={
                    Array [
                      Object {
                        "ID": "loan1",
                        "Patron ID": "patron_1",
                        "Request created": "2018-01-01 10:05",
                        "State": undefined,
                      },
                    ]
                  }
                >
                  <TableBody
                    as="tbody"
                  >
                    <tbody
                      className=""
                    >
                      <TableRow
                        as="tr"
                        cellAs="td"
                        data-test="loan1"
                        key="loan1"
                      >
                        <tr
                          className=""
                          data-test="loan1"
                        >
                          <TableCell
                            as="td"
                            key="details"
                            textAlign="center"
                          >
                            <td
                              className="center aligned"
                            >
                              <Button
                                as="button"
                                circular={true}
                                compact={true}
                                icon="eye"
                                onClick={[Function]}
                              >
                                <button
                                  className="ui circular compact icon button"
                                  onClick={[Function]}
                                >
                                  <Icon
                                    as="i"
                                    name="eye"
                                  >
                                    <i
                                      aria-hidden="true"
                                      className="eye icon"
                                    />
                                  </Icon>
                                </button>
                              </Button>
                            </td>
                          </TableCell>
                          <TableCell
                            as="td"
                            data-test="ID-loan1"
                            key="0-loan1"
                          >
                            <td
                              className=""
                              data-test="ID-loan1"
                            >
                              loan1
                            </td>
                          </TableCell>
                          <TableCell
                            as="td"
                            data-test="Request created-loan1"
                            key="1-loan1"
                          >
                            <td
                              className=""
                              data-test="Request created-loan1"
                            >
                              2018-01-01 10:05
                            </td>
                          </TableCell>
                          <TableCell
                            as="td"
                            data-test="Patron ID-loan1"
                            key="2-loan1"
                          >
                            <td
                              className=""
                              data-test="Patron ID-loan1"
                            >
                              patron_1
                            </td>
                          </TableCell>
                          <TableCell
                            as="td"
                            data-test="State-loan1"
                            key="3-loan1"
                          >
                            <td
                              className=""
                              data-test="State-loan1"
                            >
                              -
                            </td>
                          </TableCell>
                        </tr>
                      </TableRow>
                    </tbody>
                  </TableBody>
                </ResultsTableBody>
                <ResultsTableFooter
                  allRowsNumber={2}
                  columnsNumber={4}
                  showAllClickHandler={
                    Object {
                      "handler": [Function],
                      "params": null,
                    }
                  }
                  showMaxRows={1}
                >
                  <TableFooter
                    as="tfoot"
                    data-test="footer"
                    fullWidth={true}
                  >
                    <TableHeader
                      as="tfoot"
                      data-test="footer"
                      fullWidth={true}
                    >
                      <tfoot
                        className="full-width"
                        data-test="footer"
                      >
                        <TableRow
                          as="tr"
                          cellAs="td"
                        >
                          <tr
                            className=""
                          >
                            <TableHeaderCell
                              as="th"
                              colSpan={5}
                              textAlign="right"
                            >
                              <TableCell
                                as="th"
                                className=""
                                colSpan={5}
                                textAlign="right"
                              >
                                <th
                                  className="right aligned"
                                  colSpan={5}
                                >
                                  <span>
                                    Showing first 
                                    1
                                     entries of
                                     
                                    2
                                  </span>
                                  <span>
                                     
                                    <Button
                                      as="button"
                                      onClick={[Function]}
                                      size="small"
                                    >
                                      <button
                                        className="ui small button"
                                        onClick={[Function]}
                                      >
                                        Show all
                                      </button>
                                    </Button>
                                  </span>
                                </th>
                              </TableCell>
                            </TableHeaderCell>
                          </tr>
                        </TableRow>
                      </tfoot>
                    </TableHeader>
                  </TableFooter>
                </ResultsTableFooter>
              </table>
            </Table>
          </div>
        </Segment>
      </ResultsTable>
    </Error>
  </Loader>
</ItemPendingLoans>
`;
